src/replication.c

```diff
@@ -3886,11 +3875,33 @@ int connectWithPrimary(void) {
     return C_OK;
 }
 
+/* In disk-based replication, replica will open a temp db file to store the RDB file.
+ * Before entering the REPL_STATE_TRANSFER or after entering the REPL_STATE_TRANSFER,
+ * if an error occurs, we need to clean up related resources, such as closing the tmp
+ * file fd and deleting the temp file.
+ *
+ * Noted that repl_transfer_fd and repl_transfer_tmpfile should be set/unset together. */
+void cleanupTransferResources(void) {
+    if (server.repl_transfer_fd == -1) {
+        serverAssert(server.repl_transfer_tmpfile == NULL);
+        return;
+    }
+
+    serverAssert(server.repl_transfer_fd != -1 && server.repl_transfer_tmpfile != NULL);
```

Comment:
We already do the first part of the compression in the if statment above
```suggestion
    serverAssert(server.repl_transfer_tmpfile != NULL);
```